lib/utils/docker.go

```diff
@@ -1,38 +1,111 @@
 package utils
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
 	"log"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	"github.com/sdslabs/katana/configs"
 )
 
 func BuildDockerImage(_ChallengeName string, _DockerfilePath string) {
-	log.Println("Building docker image, Please wait...")
-	cmd := "docker build -t " + "harbor.katana.local/katana/" + _ChallengeName + " " + _DockerfilePath
-	if err := RunCommand(cmd); err != nil {
-		log.Printf("Error: %s\n", err)
+	buf := new(bytes.Buffer)
+	if err := Tar(_DockerfilePath, buf); err != nil {
+		log.Fatal(err, ": error tarring directory")
+		return
+	}
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	log.Println("Building Docker image, Please wait......")
+	imageBuildResponse, err := cli.ImageBuild(
+		ctx,
+		buf,
+		types.ImageBuildOptions{
+			Dockerfile: "Dockerfile",
+			Remove:     true,
+			Tags:       []string{"harbor.katana.local/katana/" + _ChallengeName},
+		},
+	)
+	if err != nil {
+		log.Fatal(err, " :unable to create image")
+		return
 	}
+
+	defer imageBuildResponse.Body.Close()
 	log.Println("Docker image built successfully")
+	DockerLogin(configs.KatanaConfig.Harbor.Username, configs.KatanaConfig.Harbor.Password)
```

Comment:
Is this required?

Reply:
yes without it , not pushing image to harbor